### PR TITLE
[CLI] Support passing of Service Account to tkn-results

### LINKED ISSUE #576: [CLI] Support the serviceaccount flag
### Feature request

The user should be able to pass serviceaccount name and namespace via the flag.
### Use case

Creating a token and using adds another step for the end-user. Instead, the passing of serviceaccount should be supported.

### PR DESCRIPTION
**Pull request by khrm**
The User can pass the flag `--sa` for serviceaccount name and `--sa-ns` for serviceaccount namespace. Latter is optional. If not given, then the namespace from the current context is assumed.

Fixes #576

# Changes

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you review them:

- [x] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [ ] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [ ] [Tested your changes locally](https://github.com/tektoncd/results/blob/main/docs/DEVELOPMENT.md) (if this is a code change)
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including functionality, content, code)
- [x] Has a kind label. You can add a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [x] Release notes block below has been updated with any user-facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings)
- [ ] Release notes contain the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
The User can pass the flag --sa for serviceaccount name and --sa-ns for serviceaccount namespace. Latter is optional. If not given, then the namespace from the current context is assumed.
```

### CODE REVIEW

**Comment by khrm** [comment]
@enarha @avinal @sayan-biswas @adambkaplan Can you review this asap? Part of milestone [v0.8.0](https://github.com/tektoncd/results/milestone/9)

**Review by khrm — COMMENTED** [review]
/test pull-tekton-results-integration-tests

File: tools/tkn-results/internal/client/client.go [hunk 1]
```diff
@@ -35,6 +35,13 @@ func NewDefaultFactory() (*ClientFactory, error) {
 	if err != nil {
 		return nil, err
 	}
+	if cfg.ServiceAccount.Name != "" && cfg.ServiceAccount.Namespace == "" {
```
Comment (avinal) [inline tools/tkn-results/internal/client/client.go:38]: If I don't pass any argument and neither provide one in the config, it gives segmentation fault. It should rather exit with a defined error.

```
./tkn-results list default     
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x15e0232]

goroutine 1 [running]:
github.com/tektoncd/results/tools/tkn-results/internal/client.NewDefaultFactory()
        /home/avinal/Documents/redhat/github.com/tektoncd/results/tools/tkn-results/internal/client/client.go:38 +0xd2
github.com/tektoncd/results/tools/tkn-results/internal/client.DefaultClient({0x1c34770, 0xc000188000}, {0x0, 0x0})
        /home/avinal/Documents/redhat/github.com/tektoncd/results/tools/tkn-results/internal/client/client.go:91 +0x2d
github.com/tektoncd/results/tools/tkn-results/cmd.Root.func1(0xc000004c00?, {0xc0001f4a40?, 0x1?, 0x1?})
        /home/avinal/Documents/redhat/github.com/tektoncd/results/tools/tkn-results/cmd/root.go:54 +0x165
github.com/spf13/cobra.(*Command).execute(0xc000004c00, {0xc0001f4a10, 0x1, 0x1})
        /home/avinal/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:891 +0x739
github.com/spf13/cobra.(*Command).ExecuteC(0xc000004900)
        /home/avinal/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:1044 +0x3bd
github.com/spf13/cobra.(*Command).Execute(...)
        /home/avinal/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:968
github.com/spf13/cobra.(*Command).ExecuteContext(0x0?, {0x1c34770?, 0xc000188000?})
        /home/avinal/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:961 +0x4a
main.main()
        /home/avinal/Documents/redhat/github.com/tektoncd/results/tools/tkn-results/main.go:9 +0x2c
```

**Review by avinal — COMMENTED** [review]
Even if I provide argument, it is not working for me. I hope I am doing it right. 
<details>
<summary>My results.yaml</summary>

```yaml
address: localhost:8080
token:
ssl:
  roots_file_path: /tmp/results.crt
  server_name_override: tekton-results-api-service.tekton-pipelines.svc.cluster.local
service_account:
  namespace:
  name:
```
 </details>
 
```
$ ./tkn-results list default --sa=tekton-results-debug --sa-ns=tekton-pipelines
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x15e0232]

goroutine 1 [running]:
github.com/tektoncd/results/tools/tkn-results/internal/client.NewDefaultFactory()
        /home/avinal/Documents/redhat/github.com/tektoncd/results/tools/tkn-results/internal/client/client.go:38 +0xd2
github.com/tektoncd/results/tools/tkn-results/internal/client.DefaultClient({0x1c34770, 0xc000048120}, {0x0, 0x0})
        /home/avinal/Documents/redhat/github.com/tektoncd/results/tools/tkn-results/internal/client/client.go:91 +0x2d
github.com/tektoncd/results/tools/tkn-results/cmd.Root.func1(0xc000004c00?, {0xc000025e00?, 0x3?, 0x3?})
        /home/avinal/Documents/redhat/github.com/tektoncd/results/tools/tkn-results/cmd/root.go:54 +0x165
github.com/spf13/cobra.(*Command).execute(0xc000004c00, {0xc000025da0, 0x3, 0x3})
        /home/avinal/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:891 +0x739
github.com/spf13/cobra.(*Command).ExecuteC(0xc000004900)
        /home/avinal/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:1044 +0x3bd
github.com/spf13/cobra.(*Command).Execute(...)
        /home/avinal/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:968
github.com/spf13/cobra.(*Command).ExecuteContext(0x0?, {0x1c34770?, 0xc000048120?})
        /home/avinal/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:961 +0x4a
main.main()
        /home/avinal/Documents/redhat/github.com/tektoncd/results/tools/tkn-results/main.go:9 +0x2c
```

File: tools/tkn-results/internal/client/client.go (hunk 1 above)
Comment (khrm) [reply tools/tkn-results/internal/client/client.go:38]: It should have been picked from the kubeconfig. Strange. Let me check.

Comment (khrm) [reply tools/tkn-results/internal/client/client.go]: This is fixed but another issue has been observed in main.

**Comment by khrm** [comment]
All flags seems to be broken.

**Comment by khrm** [comment]
@avinal Can you test this pr now?

**Review by avinal — APPROVED** [review]
Tested, working as expected. I do think that the error messages can be improved. Currently we get this error if nothing is passed or added in the config.

```
ListRecords: rpc error: code = Unauthenticated desc = permission denied
Error: rpc error: code = Unauthenticated desc = permission denied
```
We can instead return a more helpful message quoting the possible reasons. Fo now this PR is good.

**Comment by khrm** [comment]
@enarha @sayan-biswas Let's merge this pr also. @avinal Tested this.

**Review by vdemeester — APPROVED** [review]

**Pull request merged**